### DISCUSSION TITLE
Step 8: ⚙️ Set Up a Team Workflow

### DISCUSSION THREAD

**Discussion by jdmedlock** [chingu-voyages/v50-tier3-team-24#22] [General]
**__Task Description__**
- [ ] Decide on the coding standards the team must follow.
- [ ] Define the characteristics of your repo including the number of branches, branch names, & the workflow everyone must follow.
- [ ] Agree on the format of commit messages, how Pull Requests will be reviewed, and GitHub branch protection rules.
- [ ] Decide on where you’ll deploy your app and the branch to be considered as ‘production’ you’ll be deploying from.
- [ ] Decide on what meetings you’ll need each week and to save time, schedule them as recurring meetings. You might want to look ahead to [Reaching your MVP](https://github.com/chingu-voyages/Handbook/blob/main/docs/guides/voyage/voyage.md#sprint-3-5---reaching-your-mvp) to learn more about the types of meetings you’ll need.

**__Technical Considerations__**
N/a

**__Additional Considerations__**
[Setup Your Team Workflow (Chingu Handbook)](https://www.notion.so/chingucohorts/Voyages-1e528dcbf1d241c9a93b4627f6f1c809#04c66351937d48b589d835164a4ca0d5](https://github.com/chingu-voyages/Handbook/blob/main/docs/guides/voyage/voyage.md#7-setup-your-team-workflow))
[Javascript Style Guides](https://codeburst.io/5-javascript-style-guides-including-airbnb-github-google-88cbc6b2b7aa)
[Defining a Git Workflow (Chingu Handbook)](https://github.com/chingu-voyages/Handbook/blob/main/docs/resources/techresources/gitgithub.md#the-workflow)
[Git Branches (Chingu Handbook)](https://github.com/chingu-voyages/Handbook/blob/main/docs/resources/techresources/gitgithub.md#git-branches)
[Git Pull Requests (Chingu Handbook)](https://github.com/chingu-voyages/Handbook/blob/main/docs/resources/techresources/gitgithub.md#git-pull-requests)

**Member of Epic:** #16

**Comment by mnichols08** [comment]
This would be a good thing to convert to a discussion in my opinion, which can be done on the right >>>>

**Reply by kleenkanteen** [reply]
just converted it

**Comment by kleenkanteen** [comment]
@davideastmond @mnichols08 

1. for branch naming, how about we do `feat/add-do-something/3`? then the order from left to right is most meaningful to least meaningful. issue number by itself means nothing as you have to search it up. also, writing the initials seems redundant as the git log shows the author of it
2. for bugs, can we similarly do `fix/add-do-something/3`?

**Reply by kleenkanteen** [reply]
pnigng @Tandid and @joekotvas as well

**Reply by mnichols08** [reply]
1. My logic is that the main purpose of creating a branch is to propose a solution to an open issue on GitHub, so the issue number creates a symbolic link back to GitHub to maintain information about its purpose and need

The initials can be useful in the case of two people coming up with the same description such as everyone adding their name to the readme and attempting to follow a convention like this

3. I usually find it useful to think of this as `<issue-type>` which could be `fix`, `feat`, `refactor`, `delete`, etc.

**Reply by davideastmond** [reply]
At my previous job we did:
`<issueType>/<issue-number>_description-of-task`.  I think it's better if the branch name doesn't start with a number.
I also agree that the initials are probably not necessary.

**Comment by kleenkanteen** [comment]
I'm fine with David's method.

Also can we agree that the person made the PR merge the branch and delete
it after that get all the approvals? This way, the commit history shows the
correct author name next to the merge commit.

**Reply by davideastmond** [reply]
I'm ok with that.

**Comment by kleenkanteen** [comment]
Excuse the typos btw I'm using swipe to text which I sometimes don't catch
haha
